src/main/java/org/hmhb/csv/DefaultCsvService.java

```diff
@@ -0,0 +1,245 @@
+import org.springframework.stereotype.Service;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Default implementation of {@link CsvService}.
+ */
+@Service
+public class DefaultCsvService implements CsvService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultCsvService.class);
+
+    /**
+     * An injectable constructor.
+     */
+    public DefaultCsvService() {
+        LOGGER.debug("constructed");
+    }
+
+    private String handleString(
+            @Nullable String data
+    ) {
+        if (data == null) {
+            return "";
+        }
+
+        if (data.contains(",") || data.contains("\"")) {
+            return "\"" + data.replace("\"", "\"\"") + "\"";
+        } else {
+            return data;
+        }
+    }
+
+    private String flattenProgramAreas(
+            @Nonnull Set<ProgramArea> programAreas
+    ) {
+        requireNonNull(programAreas, "programAreas cannot be null");
+
+        if (programAreas.isEmpty()) {
+            return "";
+        }
+
+        StringBuilder sb = new StringBuilder();
+        for (ProgramArea programArea : programAreas) {
+            sb.append(programArea.getName()); /* None of our program areas have a comma or double quote. */
+            sb.append(',');
+        }
+
+        String toRet = sb.toString();
+
+        if (!toRet.isEmpty()) {
+            /* Strip away the trailing comma. */
+            toRet = toRet.substring(0, toRet.length() - 1);
+        }
+
+        return "\"" + toRet + "\"";
+    }
+
+    private String flattenCounties(
+            @Nonnull Set<County> counties
+    ) {
+        requireNonNull(counties, "counties cannot be null");
+
+        if (counties.isEmpty()) {
+            return "";
+        }
+
+        StringBuilder sb = new StringBuilder();
+        for (County county : counties) {
+            sb.append(county.getName()); /* None of our counties have a comma or double quote. */
+            sb.append(',');
+        }
+
+        String toRet = sb.toString();
+
+        if (!toRet.isEmpty()) {
+            /* Strip away the trailing comma. */
+            toRet = toRet.substring(0, toRet.length() - 1);
+        }
+
+        return "\"" + toRet + "\"";
+    }
+
+    @Timed
+    @Override
+    public String generateFromPrograms(
+            @Nonnull List<Program> programs,
+            @Nullable Boolean expandCounties,
+            @Nullable Boolean expandProgramAreas
+    ) {
+        LOGGER.debug(
+                "generateFromPrograms called: expandCounties={}, expandProgramAreas={}",
+                expandCounties,
+                expandProgramAreas
+        );
+        requireNonNull(programs, "programs cannot be null");
+
+        StringBuilder toReturn = new StringBuilder();
+
+        /* Print the header row. */
+        toReturn.append("org_name,org_email,org_phone,org_facebook,org_website,");
+        toReturn.append("prog_name,prog_year,prog_street,prog_city,prog_state,prog_zip,prog_coordinates,");
```

Comment:
I'll probably take out the prog_coordinates when working on #248 (import).

I don't think the coordinates will be consulted when importing orgs (I'll probably just get that info via google again).